### PR TITLE
PUD-1073|add tests for change history -serenity

### PR DESCRIPTION
**Pull request by shahnaazr**

### CODE REVIEW

File: src/test/java/cucumber/pages/ChangeHistoryPage.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
 package cucumber.pages;
 
-import net.serenitybdd.core.annotations.findby.By;
 import net.serenitybdd.screenplay.targets.Target;
 
 public class ChangeHistoryPage extends PpudPage {
 
     public static final String TITLE = TITLE_ROOT + "Change history";
     public static final Target UPLOADED_PART_A_USER = getTargetBySelector("[data-qa='PART_A_RECALL_REPORT'] [data-qa='createdByUserName']");
     public static final Target UPLOADED_PART_A_HISTORY_LINK = getTargetByDataQa("viewHistory-PART_A_RECALL_REPORT");
```
Comment (AndyNewton) [inline src/test/java/cucumber/pages/ChangeHistoryPage.java:9]: Ideally I might delete this as now un-used.

File: src/test/java/cucumber/pages/ChangeHistoryDocumentCategoryPage.java [hunk 2]
```diff
@@ -1,9 +1,11 @@
 package cucumber.pages;
 
-import net.serenitybdd.core.annotations.findby.By;
 import net.serenitybdd.screenplay.targets.Target;
 
 public class ChangeHistoryDocumentCategoryPage extends PpudPage {
 
     public static final String TITLE = TITLE_ROOT + "Uploaded document change history";
+    public static final Target MISSING_DOCUMENT_TEXT = getTargetByDataQa("missingDocumentsLabel");
+    public static final Target MISSING_DOCUMENT_DETAILS = Target.the("missing document details").locatedBy("//*[@id=\"main-content\"]/div/div/dl[@data-qa=\"missing-1-row\"]/div[2]/dd");
```
Comment (AndyNewton) [inline src/test/java/cucumber/pages/ChangeHistoryDocumentCategoryPage.java:9]: Hopefully another selector that the Cypress version can improve on considerably!

**Review by AndyNewton — APPROVED** [review]
Approved with minor comments.

File: src/test/java/cucumber/steps/NavigationSteps.java [hunk 3]
```diff
@@ -770,8 +770,11 @@ public void navigateToChangeHistory(String caseworker) {
...
                 Ensure.that(ChangeHistoryPage.UPLOADED_PART_A_USER).text().isEqualTo("Maria Badger"),
                 Ensure.that(ChangeHistoryPage.GENERATED_DOSSIER_LINK).isDisplayed(),
-                Click.on(ChangeHistoryPage.UPLOADED_PART_A_HISTORY_LINK),
-                Ensure.thatTheCurrentPage().title().isEqualTo(ChangeHistoryDocumentCategoryPage.TITLE)
+                Click.on(ChangeHistoryPage.UPLOADED_PRE_CONS_HISTORY_LINK),
+                Ensure.thatTheCurrentPage().title().isEqualTo(ChangeHistoryDocumentCategoryPage.TITLE),
+                Ensure.that(ChangeHistoryDocumentCategoryPage.MISSING_DOCUMENT_TEXT).text().isEqualTo("Missing"),
+                Ensure.that(ChangeHistoryDocumentCategoryPage.MISSING_DOCUMENT_DETAILS).text().isEqualTo("Document is missing as not received from probation yet"),
+                Ensure.that(ChangeHistoryDocumentCategoryPage.EMAIL_UPLOADED).text().isEqualTo("email.msg")
         );
```
Comment (philsegal) [inline src/test/java/cucumber/steps/NavigationSteps.java:778]: Have these changed been added to cypress changes too?

**Pull request merged**